fairroot/base/source/FairFileSourceBase.h

```diff
@@ -50,10 +50,15 @@ class FairFileSourceBase : public FairSource
 
   private:
     /**ROOT file*/
+    // TODO : Ownership should be clear. Use std::unique_ptr?
     fairroot::detail::maybe_owning_ptr<TFile> fRootFile;   //!
```

Comment:
The `maybe_owning_ptr` IS a unique_ptr.

It has a special mode where it degrades to a non-owning ptr.

So I don't see what your comment wants to say?

And we know that `maybe_owning_ptr` is nothing good in the long run (I think, it's ev en documented in that case).

Reply:
I see. When does it degrade to no-owning pointer and what controls the degradation? For me  compared to syd::unique_ptr, it's very unclear whether it's owned or not.

And I also don't quite why we have unique_ptr in other file source/sink classes but not in this class.

Reply:
> When does it degrade to no-owning pointer

Search for `non_owning` to find the cases, where it is set with a non-owning ptr. An owning-ptr can't ever be changed to a non-owning one directly. (You must `.release()` it first and then set it again as a `non_owning` one. Very, very explicitly.)

> and what controls the degradation? 

See the test suite for all the use cases: https://github.com/FairRootGroup/FairRoot/blob/dev/fairroot/tools/tests/maybe_owning.cxx

And also see the brief documentation: https://github.com/FairRootGroup/FairRoot/blob/1b5ae3e55bdbe71080c429bec96a53ef3f82a2c6/fairroot/tools/FairMemory.h#L48
It has all the relevant notes.

If you think, `maybe_owning_ptr` needs even more documentation, please write an issue or ping me on mattermost.

Reply:
Sorry, even though it's very very explicit, I still don't know whether `fRootFile` is owned by `FairFileSourceBase`. This is important because if root file is not owned, we shouldn't close it in the `FairFileSourceBase`. Then we have to define `maybe_close` to address this "maybe" situation.